src/Classes/MediaHandler.php

```diff
@@ -316,6 +316,18 @@ protected function storeFile($fileData, $disk): Media
             $origFile = file_get_contents($tmpPath . $tmpName);
             $image = Image::make($origFile);
 
+            $image_watermart_path = config('nova-media-field.image_watermark_path', null);
+            if(!is_null($image_watermart_path)) {
+                //Save image without watermark
+                $rawFilePath = $storagePath . pathinfo($newFilename, PATHINFO_FILENAME) . "-RAW." . $origExtension;
+                $file = $image->encode($origExtension, config('nova-media-field.quality', 80));
+                $disk->put($rawFilePath, $file);
+
+                // Add watermark to image
+                $watermark = Image::make($image_watermart_path);
```

Comment:
Shouldn't this be after the original file is saved in the original flow at line 342? See: https://github.com/optimistdigital/nova-media-field/blob/dbc381af185600f545f5cc487e72e61456762b0c/src/Classes/MediaHandler.php#L340-L344

Your code seems to compress and save the images twice.